### PR TITLE
.walk() should respect "detail=True"

### PR DESCRIPTION
**Pull request by JohnEmhoff**
This PR addresses two issues: passing `detail` as a kwarg to
`.walk()` fails because it internally calls `.ls()`
with both `detail=True` and whatever kwargs were passed
originally, causing `detail` to be specified twice.

The second is that `.walk()` does not respect `detail=True`
and always returns bare file names. My fix here is to just
shove the dict from `.ls()` into the return list unmodified --
this feels slightly incongruous with the non-detail version
which strips prefixes from paths, but this feels like the
right thing to do?

Anywhoo, this is a bit messy so I'm happy to clean it up
if you have any suggestions. It could also use some tests:
is there any particular way you'd like it tested? A test in
`test_local`?

### CODE REVIEW

**Comment by martindurant** [comment]
As I said elsewhere, I believe it may make sense to add this to `find()`, but not for `walk()`.

**Comment by martindurant** [comment]
OK, ok - let's have this. Can you please resolve the conflict (sorry) ?

File: fsspec/spec.py [hunk 1]
```diff
@@ -331,24 +331,40 @@ def walk(self, path, maxdepth=None, **kwargs):
...
         for info in listing:
             # each info name must be at least [path]/part , but here
             # we check also for names like [path]/part/
             name = info['name'].rstrip('/')
+            if detail:
+                to_append = info
+            else:
+                to_append = name.rsplit("/", 1)[-1]
+
             if info['type'] == 'directory' and name != path:
```
Comment (dhirschfeld) [inline fsspec/spec.py]: Is this a bug? At least in my info dict a directory is denoted with `type='dir'`:

image

Comment (martindurant) [reply fsspec/spec.py]: Should definitely be "directory" - is "dir" coming right from ftplib? Then the code in `ls` should rewrite it to "directory".

Comment (dhirschfeld) [reply fsspec/spec.py]: > *is "dir" coming right from ftplib?*

Yep - I haven't changed anything in the output. Maybe another quirk of my FTP server. I'll look to fix that up in `ls`

**Pull request closed without merging**